src/PageContent/FoodLogs/FoodLogsContainer.js

```diff
@@ -1,193 +1,114 @@
-                            {
-                                food: 'grapes',
-                                quantity: '2',
-                                unit: 'loaves'
-                            }
-                        ]
-                    },
-                    notes: '4/9, mon x4, 1-3p, none requested',
-                    raContact: {
-                        email: 'refigerator@uniongospel.org',
-                        name: 'Bob Vance',
-                        phone: '111-222-3333',
-                    },
-                    receivingAgency: { //uid replaced by info
-                        name: 'Seattle Union Gospel'
-                    },
-                    pickedUpInfo: {
-                        temperature: 29,  // in F
-                        signature: 'John Smith',
-                        timestamp: 1523174874685
-                    },
-                    deliveredInfo: {
-                        signature: 'Ellen Blake',
-                        timestamp: 1523174892769
-                    }
-                }
-            ]
+
+    async componentDidMount(){
+        const rawDeliveries = await this.fetchDeliveries();
+        this.aggrFoodLogsInfo(rawDeliveries);
+    }
+
+    async fetchDeliveries() {
+        const { account } = this.props; 
+        const agencyUid = 
+            account.accountType === AccountType.DONATING_AGENCY_MEMBER ? account.agency : account.uid; 
+        // get all deliveries' id under this agency 
+        const deliveriesIdList = [];
+        const deliveriesIdSnapshot = await deliveryIndicesRef.child(`${account.umbrella}/${agencyUid}`).once('value');
+        deliveriesIdSnapshot.forEach(dailyDeliveriesIdSnapshot =>
+            dailyDeliveriesIdSnapshot.forEach(deliveryId => {
+                deliveriesIdList.push(deliveryId.key);
+            })
+        );
+        // construct and resolve promises to fetch all deliveries
+        const deliveryPromisesList = deliveriesIdList.map(deliveryId => 
+            new Promise( async (resolve, reject) => {
+                const rawDelivery = await deliveriesRef.child(deliveryId).once('value');
+                resolve(rawDelivery.val());
+            })
+        );
+        return Promise.all(deliveryPromisesList);
+    }
+
+    async aggrFoodLogsInfo(rawDeliveries) {    
+        const filteredRawDeliveries = rawDeliveries.filter(
+            rawDelivery => rawDelivery.status === DeliveryStatus.COMPLETED
+        );
+        const agenciesInfoPromisesList = filteredRawDeliveries.map(rawDelivery => {
+            const agenciesInfoPromise = this.makeAgenciesInfoPromise( 
+                rawDelivery.delivererGroup,
+                rawDelivery.receivingAgency, 
+                rawDelivery.donatingAgency, 
+                rawDelivery.daContact,
+            );
+            return new Promise( async (resolve, reject) => {
+                const agenciesInfo = await Promise.all(agenciesInfoPromise);
```

Comment:
[Minor] Instead of having this `map` function return a list of `agenciesInfo` for every delivery and then go through another loop in `aggrDeliveries` to add into each delivery, you could also just add the `agenciesInfo` into the current delivery right here.